admin/data/sql/migrations/20190214132032.do.create-sp-refresh-score-data.sql

```diff
@@ -0,0 +1,73 @@
+CREATE PROCEDURE mtc_admin.spRefreshScoreData @checkWindowId INTEGER = NULL
+AS
+    BEGIN
+
+      IF (@checkWindowId IS NULL)
+      BEGIN
+        -- PICK CW ID WITHIN THE CURRENT CHECK PERIOD
+        SET @checkWindowId = (
+          SELECT id
+          FROM [mtc_admin].checkWindow
+          WHERE GETUTCDATE() BETWEEN checkStartDate AND adminEndDate
+        )
+      END
+      IF (@checkWindowId IS NULL)
+      BEGIN
+        -- OTHERWISE PICK THE LAST CW ID THAT OCCURRED
+        SET @checkWindowId = (
+          SELECT TOP 1 id
+          FROM [mtc_admin].checkWindow
+          WHERE GETUTCDATE() > adminEndDate
+          ORDER BY createdAt DESC
+        )
+      END
+      IF (@checkWindowId IS NULL)
+      BEGIN
+        -- RAISE AN ERROR AND EXIT WHEN NO CW ID IS DETECTED
+        RAISERROR ('NO CHECK WINDOW ID FOUND', 16, 1)
+        RETURN 1
+      END
+
+      -- CLEAR THE SCHOOL SCORE TABLE
+      DELETE FROM [mtc_admin].schoolScore
+
+      -- ADD DATA TO SCHOOL SCORE TABLE
+      INSERT INTO mtc_admin.schoolScore (checkWindow_id, school_id, score)
+        (
+          SELECT
+          ISNULL(latestPupilCheck.checkWindow_id, @checkWindowId) as checkWindowId,
+          p.school_id,
+          CAST(SUM(ISNULL(latestPupilCheck.mark, 0)) AS DECIMAL(5,2)) / COUNT(p.id) as schoolScore
```

Comment:
The divisor here shouldn't be the number of pupils, rather the number of checks.   My bad, I gave you the wrong SQL!

Reply:
@jon-shipley absolutely right!
Tacked on this commit c9c6152 please have a review